### PR TITLE
Port missing dev branch fixes from Spacefarers

### PR DESCRIPTION
**Pull request by out-of-phaze**
Same as #4530 but for stuff that has to go to dev.

### CODE REVIEW

File: code/modules/reagents/chems/chems_blood.dm [hunk 1]
```diff
@@ -70,6 +70,10 @@
 		holder.remove_reagent(type, volume)
 	. = ..()
 
+/decl/material/liquid/blood/get_reagent_color(datum/reagents/holder)
+	var/list/blood_data = REAGENT_DATA(holder, type)
+	return blood_data["blood_color"] || ..()
```
Comment (MistakeNot4892) [inline code/modules/reagents/chems/chems_blood.dm]: Data is lazylisted isn't it?

Comment (out-of-phaze) [reply code/modules/reagents/chems/chems_blood.dm]: ? yeah? so if it's null we want to return parent

Comment (MistakeNot4892) [reply code/modules/reagents/chems/chems_blood.dm]: So if it's null this will runtime.

**Review by MistakeNot4892 — CHANGES_REQUESTED** [review]
Please amend the blood data check to use LAZYACCESS() to avoid a null runtime, otherwise this PR is good to merge.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**